keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1261 @@
+policies in how cpu, memory, and devices are allocated.  Kubernetes can be viewed as being 
+an the operating system of the cloud.  Allowing specialty modules to address the use cases
+directly, rather than continuing to add complexity by continuing to modify the kubelet, 
+will allow the greatest scope of abilities while halting continued increases of complexity 
+within the core of the Kubelet.
+
+Users would like to be able to address the following use cases:
+
+* The ability to allow vendors to release vendor-specific managers for their hardware
+  and provide an interface while not having vendor-specific code within Kubelet.
+* Differentiate between types of cores and memory. <br>
+  Note - Dynamic resouce allocation does this with regular resources today.  We seek to
+  extend this ability.
+* Differentiate between different configurations of cores and memory, for instance cores
+  designated as performance versus those designated as efficiency cores 
+* Have custom plugins to optimize for particular types of workloads.  These plugins
+  may be built for performance, power reduction, cpu savings, et cetera.  <br>
+  Note:  Currently, there are very limited sets of available topology policies.  Every
+  new policy must be approved and be lockstep with the Kuberenetes release process.
+* Be able to hot-plug and test new resource managers.  
+* Be able to remove some of the complexity with current setup and, over time, reduce
+  the amount of code contained within Kubelet.  Instead, build a library with specific
+  needs.
+* Have a faster path to desired changes, without potentially impacting the core of 
+  Kubernetes with every policy change. <br>
+  Note that current solutions have been cropping up to allow for resource management
+  outside the core Kubelet. These solutions require turning off the Kubelet functionality 
+  and overriding current Kubelet allocation. We should provide a path otherwise. Many of these solutions are custom to particular companies. Many are not open source. The community should give a process to allow this functionality in core kubelet at a granular level, same as we have in many systems, such as HPC or telco.
+* Leverage additional parts from the pod spec information which could help make allocation decisions
+without the need to query the k8s controlplane.
+* Be able to do research, with minimum toil, on new policies and resource management strategies
+
+This design will also use gRPC, which is used for many other pluggable components within Kubernetes.  
+The hope is that it will also, as a side effect, allow a path to simplify the Kubelet as it is 
+today and move the complexity into external plugins.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+#### Alpha Goals
+
+1.	Provide Initial infrastructure to support CCI Driver Plugins and k8s Deployments requiring CCI Drivers after scheduling
```

Comment:
I would like not to exclude Review and possible changes in Beta.

Reply:
By initial infrastructure are you referring to the changes in kubelet to enable support for such plugins? 
Just want to explicitly point out that in addition to a very solid documentation, it would be nice to have a sample CCIDriver implementation both for testing purposes and to get people started on using it for their custom usecases. I think it would be nice to capture this explicitly in the alpha goals as well.

Reply:
Added in the goals!  Thanks for the note!